cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzurePrivateDnsZoneRegistrationEnumTest.java

```diff
@@ -0,0 +1,116 @@
+package com.sequenceiq.cloudbreak.cloud.azure;
+
+import static com.sequenceiq.cloudbreak.cloud.azure.AzurePrivateDnsZoneRegistrationEnum.AKS;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class AzurePrivateDnsZoneRegistrationEnumTest {
+    @Test
+    void testNumberOfZoneTypes() {
+        assertEquals(1, testServicesSource().count(), "Please add tests for missing enums");
+    }
+
+    @Test
+    void testZoneNamePatternCount() {
+        assertThat(AKS.getDnsZoneNamePatterns()).asList().hasSize(2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("testZoneNamePatterns")
+    void testPatterns(AzurePrivateDnsZoneRegistrationEnum serviceEnum, String testZoneName, Boolean shouldMatch) {
+        boolean zoneNameMatchedByPattern = serviceEnum.getDnsZoneNamePatterns().stream()
```

Comment:
is there any reason that we do not use `com.sequenceiq.cloudbreak.cloud.azure.validator.privatedns.AzurePrivateDnsZoneMatcherService#isZoneNameMatchingPattern` here?

Reply:
Yes: in the test code I want to reuse as little productive code as possible. This test should fail for exactly one reason, i.e. the pattern is incorrect. If the prod code for pattern matching would be reused, then there would already be 2 reasons for a test failure.